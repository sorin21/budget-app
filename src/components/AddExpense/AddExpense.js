import React, { Component } from 'react';
import { connect } from 'react-redux';

import { addExpense } from '../../actions/expensesActions';
import ExpenseForm from '../ExpenseForm/ExpenseForm';

export class AddExpense extends Component {

  daa() {
    console.log('this', this)
  }
  onSubmit = (expense) => {
    this.props.onAddExpense(expense);
    this.props.history.push('/');
    // console.log(expense)
    // console.log(this.props)
  }

  render() {
    return (
      <div>
        <h2>AddExpense with expenses</h2>
        <ExpenseForm
          // we add onSubmit here so we can use
          // the ExpenseForm in other places
          // the function is executed here
          // that is why we get back from ExpenseForm
          // the expense object
          onSubmit={this.onSubmit} />
        <hr />
      </div>
    );
  }
}


const mapDispatchProps = dispatch => {
  return {
    onAddExpense: (expense) => dispatch(addExpense(expense))
  }
}

export default connect(null, mapDispatchProps)(AddExpense);