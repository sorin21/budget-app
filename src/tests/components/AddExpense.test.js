import React from 'react';
import { shallow } from 'enzyme';

import expenses from '../../common/expenses';
import { AddExpense } from '../../components/AddExpense/AddExpense';

let onAddExpense, history, wrapper;

// run this before each test case
beforeEach(() => {
  onAddExpense = jest.fn();
  history = { push: jest.fn() };
  // render the AddExpense
  wrapper = shallow(<AddExpense history={history} onAddExpense={onAddExpense} />);
})

// AddExpense expects 2 props: onAddExpense and history
test('should render AddExpense', () => {
  expect(wrapper).toMatchSnapshot();
});

// When the form gets submitted we need to make sure that
// both spyes are called with correct information
test('should handle onSubmit', () => {
  // call the function, onSubmit,  that is passed to ExpenseForm
  // so call it with the correct data, one expense object, expense[1]
  wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])

  // we aspect that history.push spy to be called with the string '/'
  expect(history.push).toHaveBeenCalledWith('/');
  expect(onAddExpense).toHaveBeenCalledWith(expenses[1])
});