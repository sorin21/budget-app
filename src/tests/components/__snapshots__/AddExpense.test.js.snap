// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpense 1`] = `
<div>
  <h2>
    AddExpense with expenses
  </h2>
  <ExpenseForm
    onSubmit={[Function]}
  />
  <hr />
</div>
`;
